test/grouping.jl

```diff
@@ -4378,4 +4378,132 @@ end
     @test_throws ArgumentError combine(gdf, :a => (x -> [Dict('x' => 1)]) => AsTable)
 end
 
+@testset "sorting API" begin
+    # simple tests
+    df = DataFrame(x=["b", "c", "b", "a", "c"])
+    @test getindex.(keys(groupby(df, :x)), 1) == ["b", "c", "a"]
+    @test getindex.(keys(groupby(df, :x, sort=true)), 1) == ["a", "b", "c"]
+    @test getindex.(keys(groupby(df, :x, sort=NamedTuple())), 1) == ["a", "b", "c"]
+    @test getindex.(keys(groupby(df, :x, sort=false)), 1) == ["b", "c", "a"]
+    @test getindex.(keys(groupby(df, order(:x))), 1) == ["a", "b", "c"]
+    @test getindex.(keys(groupby(df, order(:x), sort=true)), 1) == ["a", "b", "c"]
+    @test_throws ArgumentError groupby(df, order(:x), sort=false)
+    @test getindex.(keys(groupby(df, order(:x), sort=NamedTuple())), 1) == ["a", "b", "c"]
+    @test getindex.(keys(groupby(df, [order(:x)])), 1) == ["a", "b", "c"]
+    @test getindex.(keys(groupby(df, [order(:x)], sort=true)), 1) == ["a", "b", "c"]
+    @test_throws ArgumentError groupby(df, [order(:x)], sort=false)
+    @test getindex.(keys(groupby(df, [order(:x)], sort=NamedTuple())), 1) == ["a", "b", "c"]
+    @test getindex.(keys(groupby(df, order(:x, rev=true))), 1) == ["c", "b", "a"]
+    @test getindex.(keys(groupby(df, order(:x, rev=true), sort=true)), 1) == ["c", "b", "a"]
+    @test getindex.(keys(groupby(df, order(:x, rev=true), sort=NamedTuple())), 1) == ["c", "b", "a"]
+    @test getindex.(keys(groupby(df, [order(:x, rev=true)])), 1) == ["c", "b", "a"]
+    @test getindex.(keys(groupby(df, [order(:x, rev=true)], sort=true)), 1) == ["c", "b", "a"]
+    @test getindex.(keys(groupby(df, [order(:x, rev=true)], sort=NamedTuple())), 1) == ["c", "b", "a"]
+    @test getindex.(keys(groupby(df, :x, sort=(;rev=true))), 1) == ["c", "b", "a"]
+    @test getindex.(keys(groupby(df, [:x], sort=(;rev=true))), 1) == ["c", "b", "a"]
+   
+    # by default sorting is not applied as range of values is wide
+    df = DataFrame(x=[2, 100, 2, 1, 100])
+    @test getindex.(keys(groupby(df, :x)), 1) == [2, 100, 1]
+    @test getindex.(keys(groupby(df, :x, sort=true)), 1) == [1, 2, 100]
+    @test getindex.(keys(groupby(df, :x, sort=NamedTuple())), 1) == [1, 2, 100]
+    @test getindex.(keys(groupby(df, :x, sort=false)), 1) == [2, 100, 1]
+    @test getindex.(keys(groupby(df, order(:x))), 1) == [1, 2, 100]
```

Comment:
Having so many equivalent ways to specify sorting does seem a bit much? Not sure if it's worth doing anything about.

Reply:
The issue is that `sort` etc. provide that many ways to specify sort order, so we cannot do anything about it.
What is the rationale behind it:
* normally people will use the "global" settings like `(rev=true,)`, which applies to all columns
* however there are cases when you want to specify sorting order per column, e.g. `[order(:x, rev=true), :y]`, where you reverse `:x` but sort on `:y` in ascending order. Therefore `order` "per column" is needed.

In general - this complexity is needed when one has several columns.